pkg/hierarchy/clusterqueue.go

```diff
@@ -16,20 +16,21 @@ limitations under the License.
 
 package hierarchy
 
-type WiredClusterQueue[CQ, Cohort nodeBase] struct {
-	cohort Cohort
+type ClusterQueue[CQ, C nodeBase] struct {
```

Comment:
It looks like this works, although maybe you were seeking consistency with Cohort?

```suggestion
type ClusterQueue[C nodeBase] struct {
```